### ISSUE TITLE
Call to a member function getRelativePath() on a non-object 

### ISSUE THREAD

**Issue by binarykitchen** [owncloud/core#17513]
### Steps to reproduce
Run occ encryption:migrate

### Expected behaviour
Do the migration and complete it successfully

### Actual behaviour
Here's the console output

```
$ sudo -u www-data php occ encryption:migrate
Reorganize system folder structure
PHP Fatal error:  Call to a member function getRelativePath() on a non-object in /var/www/owncloud/apps/files_trashbin/lib/storage.php on line 123
```

### Server configuration

```
$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 14.04.2 LTS
Release:        14.04
Codename:       trusty
```

```
$ uname -a      
Linux binarykitchen.com 3.12.27-xenU #1 SMP Thu Aug 28 23:13:07 UTC 2014 x86_64 x86_64 x86_64 GNU/Linux
```

Web server:

```
$ nginx -v
nginx version: nginx/1.8.0
```

Database:

```
$ mysql --version
mysql  Ver 14.14 Distrib 5.5.43, for debian-linux-gnu (x86_64) using readline 6.3
```

PHP version:

```
$ php -v
PHP 5.5.25-1+deb.sury.org~trusty+2 (cli) (built: May 24 2015 15:45:37) 
Copyright (c) 1997-2015 The PHP Group
Zend Engine v2.5.0, Copyright (c) 1998-2015 Zend Technologies
    with Zend OPcache v7.0.4-dev, Copyright (c) 1999-2015, by Zend Technologies
```

ownCloud version:
ownCloud 8.1.0 (stable) 

Updated from an older ownCloud or fresh install:
Fresh install via apt-get

```
$ sudo -u www-data php ./occ app:list
Enabled:
 - activity: 2.0.1
 - calendar: 0.7.1
 - documents: 0.9.0
 - encryption: 1.0.0
 - external: 1.2
 - files: 1.1.9
 - files_locking: 
 - files_pdfviewer: 0.7
 - files_sharing: 0.6.2
 - files_texteditor: 0.4
 - files_trashbin: 0.6.3
 - files_versions: 1.0.6
 - files_videoviewer: 0.1.3
 - provisioning_api: 0.2
 - templateeditor: 0.1
Disabled:
 - files_external
 - firstrunwizard
 - gallery
 - galleryplus
 - mozilla_sync
 - music
 - tasks
 - user_external
 - user_ldap
 - user_webdavauth
 - videos
```

The content of config/config.php: (Without sensitive data)

```
$ cat config.php                                                                                                                                                                                       Â î‚²Â 1Â 
<?php
$CONFIG = array (
  'instanceid' => 'xxx',
  'passwordsalt' => 'xxx',
  'datadirectory' => '/var/www/owncloud/data',
  'dbtype' => 'mysql',
  'version' => '8.1.0.8',
  'dbname' => 'owncloud',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'xxx',
  'dbpassword' => 'xxx',
  'installed' => true,
  'forcessl' => true,
  'loglevel' => 1,
  'logfile' => '/var/log/owncloud.log',
  'theme' => '',
  'maintenance' => false,
  'trusted_domains' => 
  array (
    0 => 'xxx',
  ),
  'mail_smtpmode' => 'smtp',
  'secret' => 'xxx',
  'forceSSLforSubdomains' => true,
  'mail_smtpsecure' => 'tls',
  'mail_smtphost' => 'xxx',
  'mail_smtpport' => '587',
  'mail_smtpauth' => 1,
  'mail_smtpname' => 'xxx',
  'mail_smtppassword' => 'xxx',
  'mail_smtpauthtype' => 'PLAIN',
);

```

Are you using external storage, if yes which one:
None

### Client configuration
Browser:
Firefox 39

Operating system:
Ubuntu 15.04

#### Web server error log
nginx error log is empty since a couple of days, hence irrelevant.

#### ownCloud log (data/owncloud.log)
N/A

#### Browser log
Irrelevant

**Comment by Henni** [comment]
I have the same issue.
This breaks the complete installation for me. The site only displays a blank page afterwards.

```javascript
{"reqId":"zut+yHZm74WQdYKI24Ko","remoteAddr":"","app":"PHP","message":"Call to a member function getRelativePath() on a non-object at \/var\/www\/owncloud\/apps\/files_trashbin\/lib\/storage.php#123","level":3,"time":"2015-07-09T05:52:33+00:00","method":"--","url":"--"}
```

**Comment by RealRancor** [comment]
Try to disable the files_trashbin app before doing the migration as discussed in this (probably duplicated) issue: https://github.com/owncloud/core/issues/16928

**Comment by PVince81** [comment]
CC @schiesbn

**Comment by frenchja** [comment]
@RealRancor Results with it disabled: 
```bash
sudo -u www-data php occ encryption:migrate
Reorganize system folder structure


                                                                                                                                                   
  [Doctrine\DBAL\Exception\UniqueConstraintViolationException]                                                                                     
  An exception occurred while executing 'UPDATE `oc_appconfig` SET `appid` = ? WHERE `appid` = ?' with params ["encryption", "files_encryption"]:  
  SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry 'encryption-installed_version' for key 'PRIMARY'                           
                                                                                                                                                   




                                                                                                                          
  [Doctrine\DBAL\Driver\PDOException]                                                                                     
  SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry 'encryption-installed_version' for key 'PRIMARY'  
                                                                                                                          




                                                                                                                          
  [PDOException]                                                                                                          
  SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry 'encryption-installed_version' for key 'PRIMARY'  
                                                                                                                          


encryption:migrate [user_id1] ... [user_idN]
```

**Comment by kosmoplovci** [comment]
same problem here.

**Comment by binarykitchen** [comment]
@frenchja me too, same problem here

**Comment by StAn187** [comment]
@frenchja try adding the user id of the user you want to migrate

**Comment by binarykitchen** [comment]
@StAn187 how do i get the user id and what will the encryption:migrate syntax look like with the user id?

**Comment by StAn187** [comment]
@binarykitchen Open the User list and take a look at the 2nd column :-)

**Comment by binarykitchen** [comment]
@StAn187 Nope. In the second column I see the "Full Name" ... no ID is there.

**Comment by whedro** [comment]
@binarykitchen Use your login name (First column) ...

**Comment by binarykitchen** [comment]
@whedro so, are you saying that the username equals to the user ID? both is the same?

**Comment by whedro** [comment]
@binarykitchen Yes, I try it today with my login name. In table oc_users or in other is only login name (uid). This UID is used in other tables, example for separate settings ...

in my case (ubuntu 14.04):   sudo -u www-data ./occ encryption:migrate whedro

**Comment by binarykitchen** [comment]
@whedro then next time, don't mention ID, just say username. otherwise i get confused. thanks :)

**Comment by MorrisJobke** [comment]
Regression introduced with 8.1.0 through https://github.com/owncloud/core/pull/15367

**Comment by MorrisJobke** [comment]
Fix is in #20081

**Comment by PVince81** [comment]
Fix here https://github.com/owncloud/core/pull/20081, backports here https://github.com/owncloud/core/pull/20081#issuecomment-163954836
